File: Day-31/UniqueOccurrences.js
function uniqueOccurrences(arr) {
  let occurrences = {};
  for (let i = 0; i < arr.length; i++) {
    if (occurrences[arr[i]]) {
      occurrences[arr[i]]++;
    } else {
      occurrences[arr[i]] = 1;
    }
  }
  let uniques = {};
  let countArray = Object.values(occurrences);
  for (let i = 0; i < countArray.length; i++) {
    if (uniques[countArray[i]]) {
      return false;
    } else {
      uniques[countArray[i]] = 1;
    }
  }

  return true;
}

console.log(uniqueOccurrences([1, 2, 2, 1, 1, 3]));

File: Day-31/UniqueOccurrences.py
def UniqueOccurrences(arr):
    occurrences={}
    for i in range(0, len(arr)):
        if arr[i] in occurrences:
            occurrences[arr[i]] += 1
        else:
            occurrences[arr[i]] = 1
        
    if(len(set(occurrences.values()))== len(occurrences)):
        return True
    else:
        return False

print(UniqueOccurrences([1, 2,]))